simulator/view_manager.go

```diff
@@ -149,7 +149,7 @@ func (v *ContainerView) include(o types.ManagedObjectReference) bool {
 	return v.types[o.Type]
 }
 
-func (v *ContainerView) add(root mo.Reference, seen map[types.ManagedObjectReference]bool) {
+func walk(root mo.Reference, f func(child types.ManagedObjectReference)) {
```

Comment:
Yeah, I meant to pick another name but had the same problem ;)  The usage of it may walk the whole sub-tree, examples:
- ContainerView with with Recursive option set
- EventHistoryCollector with RecursionOption Children or RecursionOption All

It isn't exported so I left the name as-is for now.